ui/components/app/confirm-page-container/confirm-page-container.component.js

```diff
@@ -229,9 +241,77 @@ export default class ConfirmPageContainer extends Component {
               ethGasPriceWarning={ethGasPriceWarning}
               hideTitle={hideTitle}
               supportsEIP1559V2={supportsEIP1559V2}
+              currentTransaction={currentTransaction}
+              nativeCurrency={nativeCurrency}
+              networkName={networkName}
+              showBuyModal={showBuyModal}
             />
           )}
-          {shouldDisplayWarning && (
+          {shouldDisplayWarning && errorKey === INSUFFICIENT_FUNDS_ERROR_KEY && (
+            <div className="confirm-approve-content__warning">
+              {currentTransaction.chainId === '0x1' ? (
+                <ActionableMessage
+                  className="actionable-message--warning"
+                  message={
+                    <Typography
+                      variant={TYPOGRAPHY.H7}
+                      align="left"
+                      margin={[0, 0]}
+                    >
+                      <I18nValue
+                        messageKey="insufficientCurrency"
+                        options={[
+                          nativeCurrency,
+                          networkName,
+                          <button
+                            key="link"
+                            type="secondary"
+                            className="confirm-approve-content__warning__link"
+                            onClick={() => showBuyModal()}
+                            style={{
+                              color: '#037dd6',
+                              padding: 0,
+                              fontSize: '12px',
+                            }}
+                          >
+                            <I18nValue messageKey="buyEth" />
```

Comment:
[REQ] Same for all instances of this component if you don't mind.

Reply:
Done.